### DISCUSSION TITLE
Most quest givers (outpost managers...) do not react to anything because NPCWaitaction

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#13960] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Most missionevents use NPCWaitaction so the NPC you talk to doesnt move, but this also means those NPCs do not react to combat in the outpost. You can shoot them or murder a NPC in front of them and they will not move. 
They also do not return to the admin module if fleeing a leak

Typical missionevent:

```xml
    <!--MANAGER MISSIONS-->
    <ScriptedEvent identifier="missionevent_cargo1" commonness="125">
      <NPCWaitAction npctag="outpostmanager" wait="true" />
      <ConversationAction text="EventText.missionevent_cargo1.c1" speakertag="outpostmanager" endeventifinterrupted="false" dialogtype="Small" />
      <MissionAction missiontag="cargomaterials" />
      <NPCWaitAction npctag="outpostmanager" wait="false" />
    </ScriptedEvent>
```

You can just remove the NPCWaitaction line from these; the outpostmanager tries staying and return in the admin module (defined in NPC prefabs). 

### Reproduction steps

1. Start campaign
2. Shoot outpost manager with revolver 6x  times
3. Manager doesnt move

4. Retry
5. Take all outpost manager missions
6. Shoot outpost manager twice
7. Outpost manager flees on 1st shot, retaliates on 2nd

_No response_

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

**Comment by lIllIIlI** [comment]
Ah, no wonder modded mission givers just stood around and did nothing while I murdered everyone in the outpost- I had chalked it up to bad programming on modders' part, but it's a vanilla issue.
Still not fixed I see.

**Comment by JeroonBosch** [comment]
> You can just remove the NPCWaitaction line from these; the outpostmanager tries staying and return in the admin module (defined in NPC prefabs).

Is this a solution you have tested? In my testing the issue doesn't seem to come from this line

**Reply by ubertpendragon** [reply]
Yes, I can just load up a campaign in vanilla, shoot the manager, he doesnt move or react at all until all missions are taken

Reload, take all the missions, then manually trigger a missionevent_xxx without WaitAction and shot the manager, he properly starts fleeing and fighting back

**Reply by JeroonBosch** [reply]
> but this also means those NPCs do not react to combat in the outpost

In my testing removing the WaitAction does not address this, they're still sitting in their admin module until you pick up their missions/quests and then start shooting, which to me means it's not really a working solution.

**Reply by ubertpendragon** [reply]
No waitaction

NPC reacts although after a certain amount of damage has been dealt

```xml    
    <ScriptedEvent identifier="missionevent_WAITACTIONTEST" commonness="100">
      <SpawnAction npcsetidentifier="outpostnpcs1" npcidentifier="watchmanseparatist" targettag="questgiver" spawnlocation="Outpost" spawnpointtag="admin"/>
      <ConversationAction text="eventtext.missionevent_piratemission1.c1" speakertag="questgiver" endeventifinterrupted="false" dialogtype="Small" />
      <MissionAction missiontag="piratemission" />
    </ScriptedEvent>
````

waitaction

NPC doesnt react

```xml  
    <ScriptedEvent identifier="missionevent_WAITACTIONTEST" commonness="100">
      <SpawnAction npcsetidentifier="outpostnpcs1" npcidentifier="watchmanseparatist" targettag="questgiver" spawnlocation="Outpost" spawnpointtag="admin"/>
      <NPCWaitAction npctag="questgiver" wait="true" />
      <ConversationAction text="eventtext.missionevent_piratemission1.c1" speakertag="questgiver" endeventifinterrupted="false" dialogtype="Small" />
      <MissionAction missiontag="piratemission" />
      <NPCWaitAction npctag="questgiver" wait="false" />
    </ScriptedEvent>
```

**Reply by JeroonBosch** [reply]
Well, they're a bit more reactive sure, but to me they're still not nearly reactive enough, and the moment you pick up their missions they instantly start shooting. This to me is odd and should also be adjusted as part of this ticket, so simply removing the WaitAction isn't enough

**Comment by lIllIIlI** [comment]
If the mission giver is a securityofficer role, then they attack you just fine after being attacked due to the inherent security's job of punishing intruders/enemies.
Thats my workaround for now.

**Comment by itchyOwl** [comment]
[Fix/npc action priority #5913](https://github.com/FakeFishGames/Barotrauma-development/pull/5913) by exposing the priority of the actions and by using 70 as the default, instead of 100, which means the bots can't do anything else.

Edit: changed the default to 100, so the existing content stays intact. WaitActions were removed where they are not needed, as suggested here. But the priority of the AI objectives from the actions can still now be adjusted between 60 and 100.

**Comment by NilanthAnimosus** [comment]
Closing as fixed and merged (coming in the next update and should already be in unstable).
